File: api-facade/src/main/java/pers/project/api/facade/controller/FacadeController.java
package pers.project.api.facade.controller;

import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;
import pers.project.api.common.model.Result;
import pers.project.api.common.model.dto.UserQuantityUsageCreationDTO;
import pers.project.api.common.model.query.ApiAdminPageQuery;
import pers.project.api.common.model.query.UserApiDigestPageQuery;
import pers.project.api.common.model.query.UserApiFormatAndQuantityUsageQuery;
import pers.project.api.common.model.vo.ApiAdminPageVO;
import pers.project.api.common.model.vo.UserApiDigestPageVO;
import pers.project.api.common.model.vo.UserApiFormatAndQuantityUsageVO;
import pers.project.api.common.util.ResultUtils;
import pers.project.api.facade.service.FacadeService;

/**
 * Facade 模块控制器
 *
 * @author Luo Fei
 * @date 2023/05/04
 */
@Validated
@RestController
@RequiredArgsConstructor
public class FacadeController {

    private final FacadeService facadeService;

    // region OpenFeign
    @PostMapping("/user/api/digest/page/result")
    public Result<UserApiDigestPageVO> getUserApiDigestPageResult
    (@Valid @RequestBody UserApiDigestPageQuery pageQuery) {
        UserApiDigestPageVO pageDTO = facadeService.getUserApiDigestPageDTO(pageQuery);
        return ResultUtils.success(pageDTO);
    }

    @PostMapping("/user/api/format/and/quantity/usage/result")
    public Result<UserApiFormatAndQuantityUsageVO> getUserApiFormatAndQuantityUsageResult
            (@Valid @RequestBody UserApiFormatAndQuantityUsageQuery query) {
        UserApiFormatAndQuantityUsageVO formatAndQuantityUsageVO
                = facadeService.getUserApiFormatAndQuantityUsageVO(query);
        return ResultUtils.success(formatAndQuantityUsageVO);
    }

    @PostMapping("/api/admin/page/result")
    public Result<ApiAdminPageVO> getApiAdminPageResult
            (@Valid @RequestBody ApiAdminPageQuery pageQuery) {
        ApiAdminPageVO pageVO = facadeService.getApiAdminPageVO(pageQuery);
        return ResultUtils.success(pageVO);
    }

    @PostMapping("/user/quantity/usage/creation/result")
    public Result<String> getUserQuantityUsageCreationResult
            (@Valid @RequestBody UserQuantityUsageCreationDTO creationDTO) {
        String usageId = facadeService.createUserQuantityUsage(creationDTO);
        return ResultUtils.success(usageId);
    }
    // endregion

}

File: api-surface/src/services/api-facade/facadeController.ts
// @ts-ignore
/* eslint-disable */
import {request} from '@umijs/max';

/** 此处后端没有提供注释 POST /api/admin/page/result */
export async function getApiAdminPageResult(
  body: API.ApiAdminPageQuery,
  options?: { [key: string]: any },
) {
  return request<API.ResultApiAdminPageVO>(`/facade/api/admin/page/result`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /quantity/usage/api/info/result */
export async function getQuantityUsageApiInfoResult(
  body: API.QuantityUsageApiInfoQuery,
  options?: { [key: string]: any },
) {
  return request<API.ResultQuantityUsageApiInfoDTO>(`/facade/quantity/usage/api/info/result`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /user/api/digest/page/result */
export async function getUserApiDigestPageResult(
  body: API.UserApiDigestPageQuery,
  options?: { [key: string]: any },
) {
  return request<API.ResultUserApiDigestPageVO>(`/facade/user/api/digest/page/result`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /user/api/format/and/quantity/usage/result */
export async function getUserApiFormatAndQuantityUsageResult(
  body: API.UserApiFormatAndQuantityUsageQuery,
  options?: { [key: string]: any },
) {
  return request<API.ResultUserApiFormatAndQuantityUsageVO>(
    `/facade/user/api/format/and/quantity/usage/result`,
    {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      data: body,
      ...(options || {}),
    },
  );
}
